src/tls/cert.h

```diff
@@ -62,17 +62,23 @@ namespace tls
         // extension) if it is set. This lets us connect to peers that present
         // certificates with IPAddress in SAN field. This is OK since we check
         // for peer CA endorsement.
-        SSL_set1_host(ssl, peer_hostname->c_str());
+        SSL_set_tlsext_host_name(ssl, peer_hostname->c_str());
```

Comment:
I think "This lets us connect to peers that present certificates with IPAddress in SAN field." only applied with `mbedtls` so this can go.

Reply:
Good point, I'll remove that.